encoding.go

```diff
@@ -40,7 +42,13 @@ func readTimestampFromEntry(data []byte) uint64 {
 
 func readKeyFromEntry(data []byte) string {
 	length := binary.LittleEndian.Uint16(data[timestampSizeInBytes+hashSizeInBytes:])
-	return string(data[headersSizeInBytes : headersSizeInBytes+length])
+	return bytesToString(data[headersSizeInBytes : headersSizeInBytes+length])
+}
+
+func bytesToString(b []byte) string {
+	bytesHeader := (*reflect.SliceHeader)(unsafe.Pointer(&b))
+	strHeader := reflect.StringHeader{Data: bytesHeader.Data, Len: bytesHeader.Len}
```

Comment:
What will happen when I read key from the cache and then underlying value is overwritten? Will I have my value?